core/drivers/stm32_uart.c

```diff
@@ -45,17 +49,26 @@ static vaddr_t loc_chip_to_base(struct serial_chip *chip)
 static void loc_flush(struct serial_chip *chip)
 {
 	vaddr_t base = loc_chip_to_base(chip);
+	uint64_t timeout = timeout_init_us(FLUSH_TIMEOUT_US);
 
 	while (!(read32(base + UART_REG_ISR) & USART_ISR_TXFE))
-		;
+		if (timeout_elapsed(timeout))
+			return;
 }
 
 static void loc_putc(struct serial_chip *chip, int ch)
 {
 	vaddr_t base = loc_chip_to_base(chip);
+	uint64_t timeout = timeout_init_us(PUTC_TIMEOUT_US);
 
-	while (!(read32(base + UART_REG_ISR) & USART_ISR_TXE_TXFNF))
-		;
+	while (!(read32(base + UART_REG_ISR) & USART_ISR_TXE_TXFNF)) {
+		if (timeout_elapsed(timeout)) {
+			if (read32(base + UART_REG_ISR) & USART_ISR_TXE_TXFNF)
```

Comment:
Why is this needed?

Reply:
Well... there is not really needed I guess. It allowed to still push a charater to the trace if timeout elapased but UART responded.
Ok to remove this test: would bring more questions than it could resolve issues.